corehq/apps/app_manager/static/app_manager/js/forms/form_view.js

```diff
@@ -170,6 +170,15 @@ $(function () {
     // Case Management > Data dictionary descriptions for case properties
     $('.property-description').popover();
 
+    $('#performance-warning-toggle').on('click', function () {
```

Comment:
This feels a little bit weird to me that we toggles the icon class based on its current class when the button is clicked not the actual collapse state.
What do you think of something like:
```
$('#warningContent').on('show.bs.collapse', function () {
    $('#performance-warning-toggle i')
        .removeClass('fa-chevron-down')
        .addClass('fa-chevron-up');
});

$('#warningContent').on('hide.bs.collapse', function () {
    $('#performance-warning-toggle i')
        .removeClass('fa-chevron-up')
        .addClass('fa-chevron-down');
});
```

Reply:
I ended up using the `collapsed` class on the toggle, which I confirmed gets added and removed automatically. The hack is that I needed to explicitly add it so that it matches that starting state of the element.

[f845770](https://github.com/dimagi/commcare-hq/pull/36588/commits/f8457704c85ce848d24b3db2e335ac067773edad)

Reply:
Just kidding, ended up going with your approach! [08d33a1](https://github.com/dimagi/commcare-hq/pull/36588/commits/08d33a11cdb8680910a90b08344a5d77cdbf95e4)